src/core/castImage.ts

```diff
@@ -0,0 +1,158 @@
+import Image from './Image.js'
+import CastImageOptions from './CastImageOptions.js'
+import PixelTypes from './PixelTypes.js'
+import IntTypes from './IntTypes.js'
+import FloatTypes from './FloatTypes.js'
+
+/**
+ * Cast an image to another PixelType or ComponentType
+ *
+ * @param {Image} image - The input image
+ * @param {CastImageOptions} options - specify the componentType and/or pixelType of the output
+ */
+function castImage (inputImage: Image, options: CastImageOptions): Image {
+  const outputImageType = { ...inputImage.imageType }
+
+  if (typeof options.pixelType !== 'undefined') {
+    outputImageType.pixelType = options.pixelType
+    if (options.pixelType === PixelTypes.Scalar && outputImageType.components !== 1) {
+      throw new Error('Cannot cast multi-component image to a scalar image')
+    }
+  }
+  if (typeof options.componentType !== 'undefined' && options.componentType !== inputImage.imageType.componentType) {
+    outputImageType.componentType = options.componentType
+  }
+
+  const outputImage = new Image(outputImageType)
+
+  outputImage.name = inputImage.name
+  outputImage.origin = Array.from(inputImage.origin)
+  outputImage.spacing = Array.from(inputImage.spacing)
+  outputImage.direction = inputImage.direction.slice()
+  outputImage.size = Array.from(inputImage.size)
```

Comment:
Could be better to do a deep copy of `inputImage` and then replace `outputImage.componentType` and `outputImage.data` but perhaps typescript wouldn't allow it. It feels like using a copy constructor would be better here as adding more properties to the Image type in the future will cause problems.
Alternatively, at least also copy `Image.metadata` which is being used for storing filenames of the dicom series, etc.

Reply:
Yes, sharing code when possible makes sense. We have to skip a full copy constructor because we do not want an extra pixel array buffer allocation and copy. I've added a copy of the `metadata`.